File: NumMeth/Demos/losssig/loss_sig.c
//#  Loss of significance demo.
//#
//# University of Cambridge Computer Laboratory - Numerical Methods Demos
//# (C) DJ Greaves, 2015.  Alan Mycroft, Silvia Breu.



#include <stdio.h>

int main()
{   int i;
    double x = 1.0/9.0;
    x *= 10;
    for (i=0; i<30; i++)
    {   printf("%e\n", x);
        x = (x - 1) * 10;   // C treats as (x-1.0) * 10.0
    }
    return 0;
}

// eof

File: NumMeth/Demos/losssig/loss_sig.ml
(*
//#  Loss of significance demo.
//#
//# University of Cambridge Computer Laboratory - Numerical Methods Demos
//# (C) DJ Greaves, 2015.  Alan Mycroft, Silvia Breu.
*)


val x_magic = 10.0/9.0
;

val x = x_magic
;

fun iterated_func x = (x-1.0) * 10.0
;

fun iterate n x = if n <= 0 then []
    else
        let val x' = iterated_func x
            val _ = print (Real.toString x' ^ "\n")
        in x :: (iterate (n-1) x')
        end     
;
val ans = iterate 30 x_magic
;

(* eof *)


